src/BinSkim.Sdk/CompilerDataLogger.cs

```diff
@@ -77,9 +81,15 @@ public static void Initialize(string instrumentationKey)
         {
             if (s_telemetryConfiguration == null && s_telemetryClient == null)
             {
-                s_sessionId = Guid.NewGuid().ToString();
-                s_telemetryConfiguration = new TelemetryConfiguration(instrumentationKey);
-                s_telemetryClient = new TelemetryClient(s_telemetryConfiguration);
+                lock (s_syncRoot)
+                {
+                    if (s_telemetryConfiguration == null && s_telemetryClient == null)
```

Comment:
>s_telemetryClient [](http://example.com/codeflow?start=60&length=17)

I don't understand the visibility of things in your class, why is s_sessionId internal but s_telemetry not?

Why is it ok to allow s_sessionId to be overridden by tests but then overwritten again if these two values are null?

Reply:
session/configuration/client can be injected using the constructor but the session is always defined when we have the telemetry enabled.

It's why we don't need to make the configuration/client internal.